File: Problem 4/problem4.py
# Project Euler Problem 4
# Tyler M Smith
# tsmith328@gatech.edu

def main():
    biggest = 0
    for i in range(999, 99, -1):
        for j in range(999, 99, -1):
            num = i * j
            numstr = str(num)
            if isPalindrome(numstr):
                if num > biggest:
                    biggest = num
    print(str(biggest) + " is the largest palindromic product of two three-digit numbers.")

def isPalindrome(s):
    i = 0
    j = len(s) - 1
    while i < j:
        if s[i] != s[j]:
            return False
        i += 1
        j -= 1
    return True

if __name__ == "__main__":
    main()

File: Problem 4/problem4.cpp
/*
 * Project Euler Problem 4
 * Tyler Smith
 * tsmith328@gatech.edu
 */

#include<iostream>
#include<string>
#include<sstream>

using std::cout;
using std::endl;

bool isPalindrome(std::string s);

namespace patch
{
    template <typename T> std::string to_string(const T& n)
    {
        std::ostringstream stm;
        stm << n;
        return stm.str();
    }
}

int main()
{
    int biggest = 0;
    for (int i = 999; i >= 100; i--)
    {
        for (int j = 999; j >= 100; j--)
        {
            int num = i * j;
            std::string numStr = patch::to_string(num);
            if (isPalindrome(numStr))
            {
                if (num > biggest)
                {
                    biggest = num;
                }
            }
        }
    }
    cout << biggest << " is the largest palindroming product of two three-digit numbers." << endl;
}

bool isPalindrome(std::string s)
{
    int i = 0;
    int j = s.length()-1;
    while (i < j)
    {
        if (s.at(i) != s.at(j))
        {
            return false;
        }
        i++;
        j--;
    }
    return true;
}